### DISCUSSION TITLE
lettuce-nioEventLoop thread is comsuming lot of CPU after migrating to spring boot 3.1.2

### DISCUSSION THREAD

**Discussion by sumit-kumar-iitm** [redis/lettuce#2883] [General]
We were running an application with spring boot 2.4.3 with Java8, recently we migrated to Spring boot 3.1.2 and Java17.
After migration we started seeing lettuce-nioEventLoop-#-# in our new relic which is taking up a lot of system CPU time. Prior to migration it was not even there on the new-relic dashboard.
Any suggestions what could have gone wrong. It is just the dependency change

**Comment by mikebell90** [comment]
So I wonder if it's the changes in Java 17 and 21 that will expose more CPUs to a containerized app? Are you running one? Java 8 also had a back and forth I believe, but you. might be running an older one that works more like the original behavior

**Reply by mikebell90** [reply]
If so, try adding `-XX:+UseContainerCpuShares` to the JVM temporarily to see if the old behavior works. And display the EXACT jvm version (not just 17, which isn't enough)

**Reply by tishun** [reply]
@sumit-kumar-iitm did you try this suggestion? Did it work?

**Reply by sumit-kumar-iitm** [reply]
We have reverted the upgrade as of now but I will try your suggestion and update on this thread. But for your reference I am sharing the most system time-consuming threads graph. You can see after we reverted lettuce-nio-EventLoop thread is negligible
<img width="906" alt="Screenshot 2024-06-12 at 6 43 46 PM" src="https://github.com/redis/lettuce/assets/4556455/9d248136-5000-46d9-958d-e7cf20b85f12">
